src/test/java/bio/terra/common/fixtures/ConnectedOperations.java

```diff
@@ -760,31 +821,65 @@ public void teardown() throws Exception {
         resetConfiguration();
 
+        // Adding try catch around each attempt because we don't want one failure to prevent us from
+        // cleaning up the rest of the items
+        // We also don't want items that couldn't be deleted to interfere with other tests
         if (deleteOnTeardown) {
             // Order is important: delete all the snapshots first so we eliminate dependencies
             // Then delete the files before the datasets
             for (String snapshotId : createdSnapshotIds) {
-                deleteTestSnapshot(snapshotId);
+                try {
+                    deleteTestSnapshot(snapshotId);
+                } catch (Exception ex) {
+                    createdSnapshotIds.remove(snapshotId);
+                    logger.info("CLEANUP ERROR! Unable to clean up snapshot {}", snapshotId);
+                }
             }
 
             for (String[] fileInfo : createdFileIds) {
-                deleteTestFile(fileInfo[0], fileInfo[1]);
+                try {
+                    deleteTestFile(fileInfo[0], fileInfo[1]);
+                } catch (Exception ex) {
+                    createdFileIds.remove(fileInfo);
+                    logger.info("CLEANUP ERROR! Unable to clean up File {}", fileInfo[0]);
+                }
             }
 
+            logger.info("{} datasets to be removed.", createdDatasetIds.size());
             for (String datasetId : createdDatasetIds) {
-                deleteTestDataset(datasetId);
+                try {
+                    deleteTestDataset(datasetId);
+                } catch (Exception ex) {
+                    removeDatasetFromTracking(datasetId);
+                    logger.info("CLEANUP ERROR! Unable to clean up dataset {}", datasetId);
+                }
             }
 
             for (String profileId : createdProfileIds) {
-                deleteTestProfile(profileId);
+                try {
+                    deleteTestProfile(profileId);
+                } catch (Exception ex) {
+                    createdProfileIds.remove(profileId);
+                    logger.info("CLEANUP ERROR! Unable to cleanup profile {}", profileId);
+                }
             }
 
             for (String bucketName : createdBuckets) {
-                deleteTestBucket(bucketName);
+                try {
+                    deleteTestBucket(bucketName);
+                } catch (Exception ex) {
+                    createdBuckets.remove(bucketName);
+                    logger.info("CLEANUP ERROR! Unable to cleanup bucket {}", bucketName);
+                }
             }
 
             for (String path : createdScratchFiles) {
-                deleteTestScratchFile(path);
+                try {
+                    deleteTestScratchFile(path);
+                } catch (Exception ex) {
+                    createdScratchFiles.remove(path);
+                    logger.info("CLEANUP ERROR! Unable to clean up scratch file {}", path);
+                }
```

Comment:
I'm not sure whether to keep this error handling or not. 
I changed the log level to info and I want tests to run a few times to see if they ever actually catch anything.